### PR TITLE
Added @JsonIgnoreProperties to all non-abstract classes

### PR DESCRIPTION
**Pull request by alebastrov**
…in order to prevent deserialization error. Also fixed sync possible issue as setting sync on holder may not be visible by other threads

### CODE REVIEW

File: dd4t-caching/src/main/java/org/dd4t/caching/providers/EHCache3Provider.java [hunk 1]
```diff
@@ -159,6 +159,7 @@ public void invalidate(final String key) {
             LOG.error("Detected undeclared null payload on element with key " + key + " at insert time!",
                     exToLogToHaveStacktraceWhoCausedIt);
             cacheElement.setNull(true);
+            //@ToDo probably bug!
```
Comment (alebastrov) [inline dd4t-caching/src/main/java/org/dd4t/caching/providers/EHCache3Provider.java:162]: take a look at this place. we have set up expiration flag and then immediately after it reset

File: dd4t-caching/src/main/java/org/dd4t/caching/impl/CacheElementImpl.java [hunk 2]
```diff
@@ -29,9 +29,9 @@
  * @author Mihai Cadariu
  */
 public class CacheElementImpl<T> implements CacheElement<T>, Serializable {
-    private boolean isExpired;
-    private boolean isNull;
-    private T payload;
+    private volatile boolean isExpired;
```
Comment (alebastrov) [inline dd4t-caching/src/main/java/org/dd4t/caching/impl/CacheElementImpl.java:32]: possible sync issue is here

File: dd4t-caching/src/main/java/org/dd4t/caching/providers/EHCache3Provider.java (hunk 1 above)
Comment (NIkonDSL) [reply dd4t-caching/src/main/java/org/dd4t/caching/providers/EHCache3Provider.java:162]: @RaimondKempees please take a look at this, it sets up the value which immediately changed

**Review by NIkonDSL — CHANGES_REQUESTED** [review]

**Comment by RaimondKempees** [comment]
This PR is too big. I don't want to add JsonIgnoreProperties everywhere, as all properties are known and explicit, so we catch new properties early (aka a new dd4t version or a new Tridion version.

The rest I am willing to consider if you trim down the PR.

**Comment by NIkonDSL** [comment]
never mind. I left SDL/RWS so no worries anymore

сб, 15 січ. 2022 р. о 14:31 RaimondKempees ***@***.***> пише:

**Pull request closed without merging**